File: src/common/user11681/wheel/extension/publish/PublishingConfig.java
package user11681.wheel.extension.publish;

import org.gradle.api.Action;

public class PublishingConfig {
    public final ExternalRepositoryConfig external = new ExternalRepositoryConfig();
    public final PublicationConfig publication = new PublicationConfig();

    public boolean enabled = true;
    public boolean local = true;

    public void external(Action<ExternalRepositoryConfig> action) {
        action.execute(this.external);
    }

    public void publication(Action<PublicationConfig> action) {
        action.execute(this.publication);
    }

    public void setPublication(boolean enabled) {
        this.publication.enabled = enabled;
    }
}

File: src/net/auoeke/wheel/extension/publish/PublishingConfig.kt
package net.auoeke.wheel.extension.publish

import groovy.lang.Closure
import org.gradle.api.Project

@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS", "unused")
class PublishingConfig(private val project: Project) {
    val external = ExternalRepositoryConfig()
    var publication = PublicationConfig()
    var enabled = true
    var local = true

    fun external(action: Closure<*>?) {
        this.project.configure(this.external, action)
    }

    fun publication(action: Closure<*>?) {
        this.project.configure(this.publication, action)
    }

    fun setPublication(enabled: Boolean) {
        this.publication.enabled = enabled
    }
}
